src/Chapter3.hs

```diff
@@ -480,6 +528,89 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data Castle = Castle
+  {
+    name :: String
+  }
+
+data IsCastle = 
+  Yes Castle |
+  No
+
+data WhichBuilding = 
+  Chruch |
+  Library
+
+data People = 
+  One Int |
+  Two Int |
+  Three Int |
+  Four Int
+
+data House = House
+  {
+    people :: People
+  }
+
+-- How to represent the Wall object?
+data City = City
+  {
+    castle :: IsCastle,
+    wall :: Bool,
+    building :: WhichBuilding,
+    houses :: [House]
+  }
+
+buildCastle :: String -> City -> City
+buildCastle newName city =
+  newCity
+  where
+    newCity :: City
+    newIsCastle :: IsCastle
+    newCastle :: Castle
+    newCastle = Castle 
+      {
+        name = newName
+      }
+    newIsCastle = Yes newCastle
+    newCity = city {castle = newIsCastle}
+
+buildHouse :: City -> House -> City
+buildHouse city house = 
+  newCity
+  where
+    newCity :: City
+    newCity = city { houses = house : (houses city)}
+
+getPeopleForHouse  :: People -> Int
+getPeopleForHouse peopl = case peopl of 
+  One _ -> 1
+  Two _ -> 2
+  Three _ -> 3
+  Four _ -> 4
+
+
+sumPeople :: [House] -> Int
+sumPeople [] = 0 
+sumPeople (x : xs) = ( getPeopleForHouse (people x))  + sumPeople(xs)
```

Comment:
You don't need parenthesis here 🙂 
Also in Haskell, functions are always separated from arguments by spaces (even if the arg is in parenthesis):

```suggestion
sumPeople (x : xs) = getPeopleForHouse (people x) + sumPeople xs
```